Fixes of random small issues

igneus opened the pull request.

igneus committed:
Fix undefined method in Line Numbers

igneus committed:
Fix undefined method in code folding

igneus commented:
Thank you very much for undertaking the upgrade.

From now on I'm using primarily your branch for my day to day music editing - more small fixes to be expected.

igneus commented on frescobaldi_app/widgets/linenumberarea.py:
```diff
@@ -57,7 +57,7 @@ def sizeHint(self):
         fm = QFontMetrics(self._textedit.font())
         text = format(self._textedit.blockCount(), 'd')
-        self._width = fm.width(text) + 3
+        self._width = fm.boundingRect(text).width() + 3
```
There are several more calls to `QFontMetrics.width()` in the codebase, but so far I didn't manage to make Frescobaldi execute them and crash, so I didn't fix them, as I hate editing code blindly (and some of them may very well be dead code which should be deleted rather than updated).

bmjcode replied:
You are a far more responsible programmer than me; I've been making most of my changes with `sed` :)

According to [here](https://doc.qt.io/qt-5/qfontmetrics-obsolete.html#width) the intended replacement is `QFontMetrics.horizontalAdvance()`, which is what I've used in other spots where this has come up. That said, I haven't worked much with QFontMetrics before. Is there a reason to use `boundingRect().width()` instead?

igneus replied:
No sound reason, just my insufficient familiarity with Qt documentation. Replaced in 98ae920

bmjcode replied:
Fair enough!

igneus committed:
Fix undefined properties in Extensions

igneus committed:
Fix undefined properties in Edit in place

igneus committed:
Fix undefined method in Edit in place

triggered by Shift+click on a textedit link in the musicview

igneus committed:
Fix undefined property in Select Matching Pair

bmjcode commented:
Thanks @igneus -- these are very helpful.

I did have a question about your `QFontMetrics` changes before I merge this (more to broaden my own knowledge than anything). See my reply to your comment above.

igneus committed:
use QFontMetrics.horizontalAdvance() for font width

it seems to be the approach preferred by the official documentation

bmjcode commented on frescobaldi_app/cursordiff.py:
I would write that constant as `QTextCursor.MoveMode.KeepAnchor` since it's a class rather than an instance member. This also makes it more visible to simple tools like `grep` and `sed` in case a future version of Qt changes the enums again.

igneus replied:
I was hesitant to change an idiom widely used throughout the codebase, but I agree that it improves discoverability both by static analysis and good old plaintext search. Will fix.

I suspect the idiom may have originally resulted from editing the codebase without autocomplete and/or automated import management.

igneus replied:
Done.

igneus committed:
Fix undefined properties in built-in Python snippets

igneus committed:
Deal with unavailability of QColorDialog.ColorDialogOptions

the code in question is called e.g. by the "Color" built-in snippet

igneus commented on frescobaldi_app/inputdialog.py:
```diff
@@ -92,12 +95,13 @@ def getColor(
         color = _savedColor
     dlg = QColorDialog(color, parent)
-    options = QColorDialog.ColorDialogOptions()
+    options = []
```
It seems that `QColorDialog.ColorDialogOptions` disappeared without replacement and or-ing the individual options is the way to go, but a double-check would be welcome. My familiarity with Qt and PyQt is really limited.

igneus replied:
Replaced with the `QColorDialog.ColorDialogOption(0)` construct.

igneus committed:
Fix type incompatibilities

In PyQt6 Qt.KeyboardModifier.* and Qt.Modifier.* aren't the same type
and cannot be combined by logical operators.
Nor can they be combined with simple integers.

(Code triggered by pressing any of the modifier keys while
in the snippet text editor.)

igneus committed:
Don't reference QTextCursor class properties via instances

while this makes the code more verbose,
it should also make the next PyQt upgrade slightly easier

igneus committed:
Fix undefined property

code called when cursor touches a bracket in the snippet text editor

igneus committed:
Better "no flag" idiom

QColorDialog.ColorDialogOption is subclass of enum.Flag and zero Flag
seems to be an idiomatic construct, used even in the official Python
documentation.

This largely reverts feb7ef0

igneus committed:
Fix missing import

triggered when opening in the Manuscript View a pdf containing links

bmjcode commented on frescobaldi_app/editinplace.py:
I appreciate that you also fixed that missing space :)

bmjcode commented on frescobaldi_app/gadgets/indenter.py:
Ugh, keyboard modifiers. You handled this right, or at least the same way I did, but the fact that this broke in the first place shows why PyQt6's use of strongly-typed enums was a mistake. The whole *point* of the `Modifier` type is not to have to spell those names out like this.

igneus replied:
`Modifier` shouldn't be a type of its own, but a collection of constants referencing `KeyboardModifier` members. Translating the idea of the C++ construct, not wrapping the C++ implementation.

bmjcode replied:
Exactly. I see why they thought the type change was a good idea; it just didn't work out so well in practice.

bmjcode commented on frescobaldi_app/inputdialog.py:
Good thinking on this one.

bmjcode commented:
Thanks again Jakub -- this has been a ton of help.
